[새기능] 어드민 분석 기능

cabbage16 opened the pull request:
## 🎫 관련 이슈
[//]: # (다음 키워드를 사용하면 해당 PR을 머지할 때 자동으로 이슈를 닫을 수 있습니다.)
[//]: # (keyword: close|closes|closed|resolve|resolves|resolved|fix|fixes|fixed)
[//]: # (예시: close #1)

close #79

<br>

## 📄 개요
[//]: # (작업 내용을 간단히 요약해서 적습니다.)
[//]: # (예시: 유저 회원가입 기능을 만들었습니다.)

> 어드민이 지원자 수, 성적 분포, 지원 전형 비율, 출신 학교 현황을 볼 수 있습니다.


<br>

## 🔨 작업 내용
[//]: # (작업 내용을 자세하게 적습니다.)
[//]: # (붙임표 "-" 을 사용해서 목록을 만듭니다.)
[//]: # (예시: 유저 회원가입 API를 만들었습니다.)

- 전형별 지원자 수 API를 만들었습니다.
- 성적 분포 API를 만들었습니다.
- 전형별 성비 API를 만들었습니다.
- 출신 학교 현황 API를 만들었습니다.



<br>

## 🏁 확인 사항
[//]: # (PR을 보내기 전 다음 사항을 확인해주세요.)
[//]: # (해당 사항을 모두 이행해야 머지할 수 있습니다.)
[//]: # (- [x] 를 사용해서 완료로 표시할 수 있습니다.)

- [x] 테스트를 완료했나요?
- [x] API 문서를 작성했나요?
- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 로그, 주석, import 등을 삭제했나요?

<br>

## 🙋🏻 덧붙일 말
[//]: # (다음 사항이 있다면 적어주세요.)
[//]: # (PR에 대한 추가 설명)
[//]: # (중점적으로 리뷰받고 싶은 부분)
[//]: # (기타 등등)

cabbage16 committed:
feat(#100): Teacher 값 nullable = false 수정
Teacher에 있는 컬럼 값이 null이 될 수 없도록 수정했습니다.

cabbage16 committed:
feat(#100): Teacher 값 nullable = false 수정
Teacher에 있는 컬럼 값이 null이 될 수 없도록 수정했습니다.

cabbage16 committed:
feat: 에러 핸들링
- statusList에 잘못된 status 값을 넣었을 때 ErrorResponse를 반환하도록 설정했습니다.

cabbage16 committed:
feat: 에러 핸들링
- statusList에 잘못된 status 값을 넣었을 때 ErrorResponse를 반환하도록 설정했습니다.

cabbage16 committed:
feat: 에러 핸들링
- statusList에 잘못된 status 값을 넣었을 때 ErrorResponse를 반환하도록 설정했습니다.

cabbage16 committed:
test(#79): UseCaseTest

cabbage16 committed:
test(#79): UseCaseTest

cabbage16 committed:
test(#79): UseCaseTest

cabbage16 committed:
refactor(#79): 불필요한 코드 정리
- 부산 출신과 타지역 출신을 찾을 때의 메서드를 통일했습니다.

cabbage16 committed:
refactor(#79): 불필요한 코드 정리
- 부산 출신과 타지역 출신을 찾을 때의 메서드를 통일했습니다.

cabbage16 committed:
test(#79): test 메서드 이름 변경
- 의미에 맞게 test 메서드 이름을 변경했습니다.

cabbage16 committed:
test(#79): test 메서드 이름 변경
- 의미에 맞게 test 메서드 이름을 변경했습니다.

cabbage16 committed:
refactor(#79): 성별 구분 로직 변경
- isFemale() 에서 isMale() 을 호출하지 않도록 변경해서 호출 횟수를 줄였습니다.

cabbage16 committed:
refactor(#79): 성별 구분 로직 변경
- isFemale() 에서 isMale() 을 호출하지 않도록 변경해서 호출 횟수를 줄였습니다.

cabbage16 committed:
test(#79): QuerySchoolStatusUseCaseTest
- QuerySchoolStatusUseCaseTest를 작성했습니다.

cabbage16 committed:
test(#79): QuerySchoolStatusUseCaseTest
- QuerySchoolStatusUseCaseTest를 작성했습니다.

cabbage16 committed:
test(#79): QueryGenderRatioUseCaseTest
- QueryGenderRatioUseCaseTest 작성했습니다.

cabbage16 committed:
test(#79): QueryGenderRatioUseCaseTest
- QueryGenderRatioUseCaseTest 작성했습니다.

cabbage16 committed:
Merge branch 'feat/#79' of https://github.com/Bamdoliro/marubase into feat/#79

cabbage16 committed:
refactor(#79): 필요없는 import 문 삭제

cabbage16 committed:
refactor(#79): 필요없는 import 문 삭제

gimhanul commented on src/main/java/com/bamdoliro/maru/application/analysis/QueryGenderRatioUseCase.java:
```diff
@@ -71,6 +71,6 @@ private boolean isMale(Form form) {
 
     private boolean isFemale(Form form) {
-        return !isMale(form);
+        return form.getApplicant().getGender().equals(Gender.FEMALE);
```
enum 비교는 equals보다 ==을 사용하는 게 안전하므로 컨벤션으로 들고갔으면 좋겠습니다!

cabbage16 committed:
fix(#100): 고입검정 입학 요강에 맞게 nullable 수정

- 고입검정 학생은 Education 에서 졸업구분과 합격 년도를 제외한 모든 값이 존재하지 않으므로 nullable = true 로 변경했습니다.
- @Embeddable 객체는 모든 필드가 null 이면 객체 자체가 null 이 되어 조회 시 NullPointerException 이 발생하는 문제를 설정을 통해 null 이 아니게 되도록 변경했습니다.

 엔티티 클래스를 수정하고 서버를 다시 돌렸는데 컬럼에 수정 사항이 적용이 안되어서 결국 직접 컬럼들의 속성을 수정했습니다. 왜 이런 현상이 발생하는지 답변 부탁드립니다...

cabbage16 committed:
test(#100): 컬럼 변경사항에 따라 설명과 FormFixture 수정
- 학교 관련 필드에도 없으면 null 이라는 설명을 추가했어요.
- 고입검정 학생 요청 예시에서 school 관련 필드도 null 로 요청하도록 변경했어요.

cabbage16 committed:
Merge branch 'fix/#100' of https://github.com/Bamdoliro/marubase into fix/#100

cabbage16 committed:
fix: git conflict 해결

cabbage16 committed:
perf(#100): NullPointerException 방지
- toString() 메서드에서 null 을 반환하는 것은 NPE 를 발생시킬 수 있으므로, value 가 null 이면 빈 문자를 반환하고 이를 DTO 에서 처리하도록 수정했습니다.

cabbage16 committed:
Merge pull request #101 from Bamdoliro/fix/#100

[버그] 고입 검정 학생의 학교, 선생님 정보에 대한 null 값 처리

cabbage16 committed:
Merge remote-tracking branch 'origin/feat/#79' into feat/#79

cabbage16 committed:
perf(#79): Enum 비교 시 == 연산자 사용
- Enum 비교 시 equals() 를 사용하면 NullPointerException 이 발생할 수 있으므로 == 으로 수정했습니다.
